fc/jira/issue.py

```diff
@@ -0,0 +1,141 @@
+        self.description = json['fields']['description']
+        self.id = json['key']
+        self.url = self.base_url.format(json['key'])
+        self.type = json['fields']['issuetype']['name']
+        self.state = json['fields']['status']['name']
+        self.auth = auth
+
+        return self
+
+    def from_args(self, title: str, description: str, auth: Auth):
+        self.title = title
+        self.description = description
+        self.auth = auth
+
+        return self
+
+    def create(self):
+        json = {
+            'fields': {
+                'project': {
+                    'key': 'QPPFC'
+                },
+                'summary': self.title,
+                'description': self.description,
+                'components': [{
+                    'name': 'Foundational'
+                }]
+            }
+        }
+
+        self._extra_json_for_create(json)
+
+        response = requests.post(self.api_url, json=json,
+                                 auth=HTTPBasicAuth(self.auth.username(), self.auth.password()))
+        response.raise_for_status()
+
+        response_json = response.json()
+
+        self.id = response_json['key']
+        self.url = self.base_url.format(self.id)
+
+        return self.id, self.url
+
+    def _extra_json_for_create(self, existing_json: dict):
+        raise NotImplementedError
+
+    def type_str(self) -> str:
+        raise NotImplementedError
+
+    @classmethod
+    def _get_issue(cls, issue_id: str, auth: Auth) -> dict:
+        response = requests.get(cls.api_url + issue_id,
+                                auth=HTTPBasicAuth(auth.username(), auth.password()))
+        response.raise_for_status()
+
+        return response.json()
+
+    def _get_active_sprint_id_of_issue(self, issue_id: str) -> Optional[int]:
+        issue = self._get_issue(issue_id, self.auth)
+
+        sprint_list = issue['fields'][self.issue_assigned_sprint_field]
+
+        if sprint_list is None:
+            return None
+
+        active_sprint = sprint_list[-1]
+        id_begin_token = 'id='
+        id_begin = active_sprint.find(id_begin_token) + len(id_begin_token)
+        id_end = active_sprint.find(',', id_begin)
+
+        return int(active_sprint[id_begin:id_end])
+
+    def _get_transition_dict(self) -> dict:
+        raise NotImplementedError
+
+    @classmethod
+    def get_issue(cls, issue_id: str, auth: Auth) -> 'Task':
+        from .backlog_task import BacklogTask
+        from .triage_task import TriageTask
+        from .el_task import ElTask
+
+        issue_json = {}
+        task = None
+
+        try:
+            issue_json = cls._get_issue(issue_id, auth)
+        except HTTPError as exception:
+            raise TaskException('Invalid issue key {}'.format(issue_id)) from exception
+
+        type = issue_json['fields']['issuetype']['name']
+        if type != 'Task' and type != 'Triage Task':
+            raise TaskException('Invalid issue type {}'.format(type))
```

Comment:
My idea was to allow `fcli` to comment on any Jira issue, hence making the extra layer of abstraction with the `Issue` class.

So, instead of raising a `TaskException`, let's create an `Issue`!

Something like...
`task = cls().from_json(issue_json, auth)`
perhaps?
Or I wonder if you can do `task = Issue.from_json(issue_json, auth)`
Or I wonder if you can do `task = 'Issue'.from_json(issue_json, auth)`
Also `task` should be made `issue`

Reply:
I updated, let me know if this is what you had in mind, @halprin. Will clean up as necessary.

Reply:
Wait, put it in the wrong place! One moment.

Reply:
Disregard, I guess it is good then. Maybe we review later and get this pushed through.